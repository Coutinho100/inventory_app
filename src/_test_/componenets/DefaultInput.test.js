import React from 'react';
import DefaultInput from '../../components/Component/DefaultInput';
import { View } from 'react-native';
import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { TextInput } from 'react-native';
import renderer from 'react-test-renderer';
configure({ adapter: new Adapter() });


describe("<Default input/>", () => {
    let wrapper, mockClick, props;
    beforeEach(() => {
        props = {
            touched: true,
            valid: true
        }

    });
    test('renders correctly', () => {
        const tree = renderer.create(<DefaultInput />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Default input should render textinput', () => {

        const wrapper = mount(<DefaultInput />);
        expect(wrapper.hasClass('invalid'));
        expect(wrapper.contains(<TextInput />));

    });

});






