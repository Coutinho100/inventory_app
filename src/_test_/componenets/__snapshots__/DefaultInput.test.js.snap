// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Default input/> renders correctly 1`] = `
<TextInput
  allowFontScaling={true}
  rejectResponderTermination={true}
  style={
    Array [
      Object {
        "borderColor": "#eee",
        "borderWidth": 1,
        "marginBottom": 8,
        "marginTop": 8,
        "padding": 5,
        "width": "100%",
      },
      undefined,
      null,
    ]
  }
  underlineColorAndroid="transparent"
/>
`;
